edxval/api.py

```diff
@@ -317,26 +318,46 @@ def get_url_for_profile(edx_video_id, profile):
 def _get_videos_for_filter(
         video_filter,
         sort_field=None,
-        sort_dir=SortDirection.asc
+        sort_dir=SortDirection.asc,
+        **kwargs
 ):
     """
     Returns a generator expression that contains the videos found, sorted by
     the given field and direction, with ties broken by edx_video_id to ensure a
     total order.
     """
-    videos = Video.objects.filter(**video_filter)
+    videos_qs = Video.objects.filter(**video_filter)
+    page_no = int(kwargs.pop("page", 1))
+    page_size = int(kwargs.pop("page_size", 20))
+
     if sort_field:
         # Refining by edx_video_id ensures a total order
-        videos = videos.order_by(sort_field.value, "edx_video_id")
+        videos_qs = videos_qs.order_by(sort_field.value, "edx_video_id")
         if sort_dir == SortDirection.desc:
-            videos = videos.reverse()
```

Comment:
I'll write a test that'll make sure that videos are reverse and paginated as expected.